.vscode/c_cpp_properties.json

```diff
@@ -9,8 +9,8 @@
                 "/src/source/**/include"
             ],
             "compilerPath": "/usr/bin/gcc",
-            "cStandard": "c17",
-            "cppStandard": "gnu++17",
+            "cStandard": "c23",
+            "cppStandard": "gnu++20",
```

Comment:
I was expecting to use C++20 features, but it's no longer the case. We can revert or upgrade, no strong preference on my end

Reply:
fair, but we should consider whether we want to do this at some point during this breaking change. (there is still some value in staying back, so that's ok on my end too)

Reply:
Yeah we can do it, but I would just make a separate PR for this